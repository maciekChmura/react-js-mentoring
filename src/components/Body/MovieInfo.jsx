import React from 'react';
import styled from 'styled-components';

const MovieInfoWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;

const Title = styled.p`
  font-size: 14px;
  color: black;
  margin: 5px;
`;

const Year = styled.p`
  color: black;
  font-size: 14px;
  margin: 5px;
`;

const MovieInfo = props => {
  const { title, release_date } = props;
  return (
    <MovieInfoWrapper>
      <Title>{title}</Title>
      <Year>{release_date.slice(0, 4)}</Year>
    </MovieInfoWrapper>
  );
};

export default MovieInfo;
