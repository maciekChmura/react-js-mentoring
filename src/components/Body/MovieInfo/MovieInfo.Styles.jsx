import styled from 'styled-components';

export const MovieInfoWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;

export const Title = styled.p`
  font-size: 14px;
  color: black;
  margin: 5px;
`;

export const Year = styled.p`
  color: black;
  font-size: 14px;
  margin: 5px;
`;
