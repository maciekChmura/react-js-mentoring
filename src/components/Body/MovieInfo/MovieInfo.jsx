import React from 'react';
import { MovieInfoWrapper, Title, Year } from './MovieInfo.Styles';

const MovieInfo = ({ title, release_date }) => {
  const releaseDate = release_date.slice(0, 4);
  return (
    <MovieInfoWrapper>
      <Title>{title}</Title>
      <Year>{releaseDate}</Year>
    </MovieInfoWrapper>
  );
};

export default MovieInfo;
