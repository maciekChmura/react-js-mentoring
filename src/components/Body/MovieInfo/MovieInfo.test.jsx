import React from 'react';
import { shallow } from 'enzyme';

import MovieInfo from './MovieInfo';

describe('MovieInfo', () => {
  const title = 'Finding Nemo';
  const release_date = '2003-05-30-';
  it('should render correctly', () => {
    const component = shallow(
      <MovieInfo title={title} release_date={release_date} />
    );

    expect(component).toMatchSnapshot();
  });
});
