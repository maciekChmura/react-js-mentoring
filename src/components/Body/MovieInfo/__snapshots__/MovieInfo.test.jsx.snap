// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieInfo
    release_date="2003-05-30-"
    title="Finding Nemo"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          Finding Nemo
        </ForwardRef>,
        <ForwardRef>
          2003
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Finding Nemo",
        },
        "ref": null,
        "rendered": "Finding Nemo",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "MovieInfoStyles__Title-sc-16dmz1l-1",
            "isStatic": true,
            "rules": Array [
              "font-size:14px;color:black;margin:5px;",
            ],
          },
          "displayName": "MovieInfoStyles__Title",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "MovieInfoStyles__Title-sc-16dmz1l-1",
          "target": "p",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "2003",
        },
        "ref": null,
        "rendered": "2003",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "MovieInfoStyles__Year-sc-16dmz1l-2",
            "isStatic": true,
            "rules": Array [
              "color:black;font-size:14px;margin:5px;",
            ],
          },
          "displayName": "MovieInfoStyles__Year",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "MovieInfoStyles__Year-sc-16dmz1l-2",
          "target": "p",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "MovieInfoStyles__MovieInfoWrapper-sc-16dmz1l-0",
        "isStatic": true,
        "rules": Array [
          "display:flex;flex-direction:row;justify-content:space-between;",
        ],
      },
      "displayName": "MovieInfoStyles__MovieInfoWrapper",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "MovieInfoStyles__MovieInfoWrapper-sc-16dmz1l-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            Finding Nemo
          </ForwardRef>,
          <ForwardRef>
            2003
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Finding Nemo",
          },
          "ref": null,
          "rendered": "Finding Nemo",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "MovieInfoStyles__Title-sc-16dmz1l-1",
              "isStatic": true,
              "rules": Array [
                "font-size:14px;color:black;margin:5px;",
              ],
            },
            "displayName": "MovieInfoStyles__Title",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "MovieInfoStyles__Title-sc-16dmz1l-1",
            "target": "p",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "2003",
          },
          "ref": null,
          "rendered": "2003",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "MovieInfoStyles__Year-sc-16dmz1l-2",
              "isStatic": true,
              "rules": Array [
                "color:black;font-size:14px;margin:5px;",
              ],
            },
            "displayName": "MovieInfoStyles__Year",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "MovieInfoStyles__Year-sc-16dmz1l-2",
            "target": "p",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "MovieInfoStyles__MovieInfoWrapper-sc-16dmz1l-0",
          "isStatic": true,
          "rules": Array [
            "display:flex;flex-direction:row;justify-content:space-between;",
          ],
        },
        "displayName": "MovieInfoStyles__MovieInfoWrapper",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "MovieInfoStyles__MovieInfoWrapper-sc-16dmz1l-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
